susemanager-utils/susemanager-sls/src/states/appstreams.py

```diff
@@ -0,0 +1,101 @@
+'''
+Salt Custom State for configuring AppStreams
+
+This custom state provides functionality for configuring (enabling and disabling)
+AppStreams modules. Its 'configured' custom state offers a solution for managing
+AppStreams by facilitating both the enabling and disabling operations simultaneously.
+
+Note: When invoking the 'configured' state, the disable operation is executed first,
+ensuring a seamless transition and minimizing potential errors. This sequential approach
+is especially beneficial in scenarios where it's necessary to disable a previous version
+of the same AppStream module before enabling a newer version - In that case, the enabling
+operation would fail if the previous version were not disabled. Moreover, in the
+event of an error during the enable operation, the previously disabled modules are
+automatically reenabled.
+'''
+
+__virtualname__ = "appstreams"
+
+def __virtual__():
+    """
+    Only if the appstreams module is loaded
+    """
+    return (
+        __virtualname__
+        if ("appstreams.enable" in __salt__
+            and "appstreams.disable" in __salt__
+            and "appstreams.get_enabled_modules" in __salt__)
+        else (False, "Module appstreams: appstreams module can't be loaded")
+    )
+
+def configured(name, appstreams_to_enable, appstreams_to_disable):
+    '''
+    Change the state of AppStreams.
+
+    name (str)
+        The name of the state
+
+    appstreams_to_enable (list):
+        List of appstreams to enable in the format module_name:stream
+
+    appstreams_to_disable (list):
+        List of names of appstreams to disable
+    '''
+    ret = {"name": name, "result": True, "changes": { "ret": {} }, "comment": ""}
+
+    if __opts__["test"]:
+        return _test_mode(ret, appstreams_to_enable, appstreams_to_disable)
+
+    if appstreams_to_disable:
+        disabled_success, disabled_comment, disabled_changes = _do_operation(name, appstreams_to_disable, "appstreams.disable")
+        if not disabled_success:
+            ret["result"] = False
+            ret["comment"] = disabled_comment
+            return ret
+
+        ret["changes"]["ret"]["disabled"] = disabled_changes["disabled"]
```

Comment:
If I follow the code correctly, this information can be wrong. When L63 re-enables a module that was disabled here, will the module be listed in both "disabled" and "enabled"?